src/components/ChatRoom.tsx

```diff
@@ -103,9 +105,33 @@ function Editor({ roomId, message, onClose, ...rest }: {
     message ? message.markdown : "");
   const [saving, setSaving] = useState<boolean>(false);
   const utils = trpcNext.useContext();
-  
-  const insertSnippet = (snippet: string) => {
-    setMarkdown(prev => prev + snippet);
+  const direction: "row" | "column" | undefined = useBreakpointValue({
+    base: 'column' as const,
+    md: 'row' as const,
+  });
+  const textareaRef = useRef<HTMLTextAreaElement>(null);
+  const insertSnippet = (markdown: string, cursorPos: number, snippet: string) => 
+  {
+    const before = markdown.substring(0, cursorPos);
+    const after = markdown.substring(cursorPos);
+    return before + snippet + after;
+  };
+
+  const handleSelectChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    const textarea = textareaRef.current;
+    if (textarea) {
+      const cursorPos = textarea.selectionStart; 
+      const snippet = snippets.find(snippet => 
+        snippet.text === e.target.value)?.text ?? '';
+      const updatedMarkdown = insertSnippet(markdown, cursorPos, snippet);
+      setMarkdown(updatedMarkdown); 
+
+      setTimeout(() => {
+        const newCursorPos = updatedMarkdown.length;
```

Comment:
Why do you need to update cursor position? Why can't leave its position as is?

Reply:
The cursor position will be updated to where the user wants to insert the snippet, and then it will move to the end of the updated textarea so the user can continue typing.

Reply:
If the user wants to insert the snippet at a certain position, it's more natural for the cursor stays at the end of the snippet than the end of the whole text, because it's more consistent with the cursor's expected behavior (right after the last inserted character).